src/monodroid/jni/monodroid-glue-external.hh

```diff
@@ -0,0 +1,67 @@
+// Dear Emacs, this is a -*- C++ -*- header
+#ifndef __MONODROID_GLUE_EXTERNAL_HH
+#define __MONODROID_GLUE_EXTERNAL_HH
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <jni.h>
+
+#include "monodroid.h"
+#include "logger.hh"
+#include "timing.hh"
+#include "xamarin_getifaddrs.h"
+
+namespace xamarin::android
+{
+	class MonodroidExternalAPI
+	{
+	public:
+		static constexpr char INIT_FUNCTION_NAME[] = "_monodroid_init_external_api";
+
+	public:
+		virtual mono_bool monodroid_get_network_interface_up_state (const char *ifname, mono_bool *is_up) = 0;
```

Comment:
Should this type have a public virtual destructor?

Reply:
In theory, yes, in practice, no. It's a pure interface class, no code here ever runs. And the only descendant is our Impl class, from which no further classes can be derived.

Reply:
Depending on how your "can we have a `_fini` function/`__attribute__((destructor))`-based cleanup code in `libxa-internal-api.so`?" investigation goes, we *may* want a virtual destructor, so that when `libxa-internal-api.so!internal_calls` is destructed/cleaned up, it'll do the correct thing.

For example, we could (maybe, perhaps, at least in spirit?) make `internal_calls` a `std::unique_ptr<MonoAndroidInternalCalls>`, and when the library is unloaded -- assuming it's unloaded or otherwise cleaned up! -- then the `MonoAndroidInternalCalls` destructor would be executed.

Needs more investigation, to be sure.

Reply:
From a very quick scan of the Mono source, I don't think Mono calls `dlclose` on the handle it gets from us during runtime, it closes the handle(s) only as part of the app shutdown cleanup. So I'm not sure we'd gain anything from `_fini` or `std::unique_ptr`.